### PR TITLE
Correct the order of updating values for "tab" dropdown

### LINKED ISSUE #209: Header data dropdown didn't update input values of the dropdown in data tab
The display selections in both "tab" dropdown (project, folder, template) and "header" dropdown are synced, but the values in "tab" dropdown are not updated immediately (below). It can cause to submit to wrong folders. Looking into it today.

Screen Shot 2021-10-08 at 12 44 10 PM

### PR DESCRIPTION
**Pull request by rrchai**
- fixes #209 
- use scss code to replace r code chunk for truncating long folder names using ellipsis in `server.R`

### CODE REVIEW

File: server.R [hunk 1]
```diff
@@ -155,14 +126,36 @@ shinyServer(function(input, output, session) {
...
+      if (x == "dropdown_") {
+        project_synID <<- projectID
+        datatype_list$folders <<- folder_list
+      }
+
+      req(input$update_confirm)
+      updateSelectInput(session, "dropdown_folder", selected = input[["header_dropdown_folder"]])
+    })
+  })
+
```
Comment (rrchai) [inline server.R:158]: I have forced it to update folder selection after the folder list updated here

**Comment by rrchai** [comment]
@milen-sage 

The issue happened when both "project" and "folder" are changed in the "header" dropdown menu. The values of folder in the "tab" dropdown will not always synced with the selection in the "header" dropdown. It will show the first folder in the dropdown list. 

The reason to cause this is because:
1. click "confirm" button will update both "project" and "folder" dropdown
2. updating "project" dropdown will trigger to update the "folder" dropdown menu list and it will auto assign the "folder" value with the first item in the folder list
3. this update may overwrite the update from "folder" dropdown in header

File: server.R [hunk 2]
```diff
@@ -125,16 +106,6 @@ shinyServer(function(input, output, session) {
     }
   })
 
-  lapply(datatypes, function(x) {
```
Comment (milen-sage) [inline server.R:128]: @rrchai did you mean to drop this block? I don't think I see it in the updated code.

File: server.R (hunk 1 above)
Comment (milen-sage) [reply server.R:158]: @rchai, I see - so lines 154-156 essentially take whatever the user selected in the header tab dropdown, and ensure the 'main' dropdowns (on the first tab of the app) are set to these values?

File: server.R (hunk 2 above)
Comment (rrchai) [reply server.R:128]: yes! This chunk is replaced by scss code in `_header.scss`

File: server.R [hunk 3]
```diff
@@ -154,13 +126,38 @@ shinyServer(function(input, output, session) {
...
+      if (x == "dropdown_") {
+        project_synID <<- projectID
+        datatype_list$folders <<- folder_list
+      }
+
+      req(isUpdateFolder() == TRUE)
+      # sync with header dropdown
+      updateSelectInput(session, "dropdown_folder", selected = input[["header_dropdown_folder"]])
+      isUpdateFolder(FALSE)
+    })
```
Comment (rrchai) [inline server.R:160]: sorry, just fixed some issues from the earlier codes.

> @rchai, I see - so lines 156-159 essentially take whatever the user selected in the header tab dropdown, and ensure the 'main' dropdowns (on the first tab of the app) are set to these values?

yes, these lines are used to force the "main" folder dropdown to update again with the same value in the "header" folder dropdown, if users click "confirm" button.

**Pull request merged**